x-pack/plugins/reporting/public/lib/stream_handler.ts

```diff
@@ -132,30 +147,38 @@ export class ReportingNotifierStreamHandler {
   /*
    * An observable that finds jobs that are known to be "processing" (stored in
    * session storage) but have non-processing job status on the server
+   *
+   * Public for purposes of testing
    */
-  public findChangedStatusJobs(storedJobs: JobId[]): Rx.Observable<JobSummarySet> {
-    return Rx.from(this.apiClient.findForJobIds(storedJobs)).pipe(
-      map((jobs) => {
-        const completedJobs: JobSummary[] = [];
-        const failedJobs: JobSummary[] = [];
-        const pending: JobId[] = [];
-
-        // add side effects to storage
-        for (const job of jobs) {
-          const { id: jobId, status: jobStatus } = job;
-          if (storedJobs.includes(jobId)) {
-            if (jobStatus === JOB_STATUS.COMPLETED || jobStatus === JOB_STATUS.WARNINGS) {
-              completedJobs.push(getReportStatus(job));
-            } else if (jobStatus === JOB_STATUS.FAILED) {
-              failedJobs.push(getReportStatus(job));
-            } else {
-              pending.push(jobId);
-            }
+  public findChangedStatusJobs(previousPending: JobId[]): Rx.Observable<JobSummarySet> {
+    return Rx.from(this.apiClient.findForJobIds(previousPending)).pipe(
+      mergeMap(async (jobs) => {
+        const newCompleted: JobSummary[] = [];
+        const newFailed: JobSummary[] = [];
+        const newPending: JobId[] = [];
+
+        for (const pendingJobId of previousPending) {
```

Comment:
main change is here: loop through the set of stored job IDs rather than loop through the job IDs returned in the search